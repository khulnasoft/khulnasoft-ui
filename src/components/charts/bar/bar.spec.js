import { shallowMount } from '@vue/test-utils';
import { createMockChartInstance } from '~helpers/chart_stubs';
import { expectHeightAutoClasses } from '~helpers/chart_height';
import Chart from '../chart/chart.vue';
import ChartTooltip from '../tooltip/tooltip.vue';
import BarChart from './bar.vue';

let mockChartInstance;

jest.mock('echarts', () => ({
  getInstanceByDom: () => mockChartInstance,
}));

const defaultChartProps = {
  xAxisTitle: 'Pushes per day',
  xAxisType: 'value',
  yAxisTitle: 'User',
  data: {
    Office: [
      [100, 'Scranton Strangler'],
      [300, 'Dwight'],
    ],
  },
};

describe('Bar chart component', () => {
  let wrapper;

  const findChart = () => wrapper.findComponent(Chart);
  const findChartTooltip = () => wrapper.findComponent(ChartTooltip);

  const getOptions = () => findChart().props('options');
  const emitChartCreated = () => findChart().vm.$emit('created', mockChartInstance);

  const createComponent = (props = {}) => {
    wrapper = shallowMount(BarChart, {
      propsData: { ...defaultChartProps, ...props },
    });
  };

  beforeEach(() => {
    mockChartInstance = createMockChartInstance();
  });

  describe('when mounted', () => {
    beforeEach(() => {
      createComponent();
    });

    it('should render chart with axis and series', () => {
      const chart = findChart();

      expect(chart.props('options')).toMatchSnapshot();
    });

    it('should not emit created', () => {
      expect(wrapper.emitted('created')).toBeUndefined();
    });
  });

  describe('when mounted and chart has created', () => {
    beforeEach(() => {
      createComponent();
      emitChartCreated();
    });

    it('emits "created" when onCreated is called', () => {
      expect(wrapper.emitted('created')).toEqual([[mockChartInstance]]);
    });

    it('long Y axis labels are ellipsized', () => {
      const { yAxis: { axisLabel: { formatter } } = {} } = getOptions();

      defaultChartProps.data.Office.map(([, name]) => expect(formatter(name)).toMatchSnapshot());
    });

    it('configures chart tooltip', () => {
      expect(findChartTooltip().props()).toMatchObject({
        chart: mockChartInstance,
        useDefaultTooltipFormatter: false,
      });
    });
  });

  describe('height', () => {
    expectHeightAutoClasses({
      createComponent,
      findContainer: () => wrapper,
      findChart,
    });
  });
});
