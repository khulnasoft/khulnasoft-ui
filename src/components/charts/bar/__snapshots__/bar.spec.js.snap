// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar chart component when mounted and chart has created long Y axis labels are ellipsized 1`] = `"Scra..."`;

exports[`Bar chart component when mounted and chart has created long Y axis labels are ellipsized 2`] = `"Dwight"`;

exports[`Bar chart component when mounted should render chart with axis and series 1`] = `
Object {
  "grid": Object {
    "bottom": 44,
    "left": 64,
    "right": 32,
    "top": 16,
  },
  "legend": Object {
    "show": false,
  },
  "series": Array [
    Object {
      "barMaxWidth": "50%",
      "data": Array [
        Array [
          100,
          "Scranton Strangler",
        ],
        Array [
          300,
          "Dwight",
        ],
      ],
      "emphasis": Object {
        "itemStyle": Object {
          "color": "rgba(97, 122, 226, 0.4)",
        },
      },
      "itemStyle": Object {
        "borderColor": "#617ae2",
        "borderWidth": 1,
        "color": "rgba(97, 122, 226, 0.2)",
      },
      "name": "Office",
      "stack": "chart",
      "type": "bar",
    },
  ],
  "xAxis": Object {
    "axisLabel": Object {
      "formatter": [Function],
    },
    "axisLine": Object {
      "show": false,
    },
    "name": "Pushes per day",
    "nameLocation": "center",
    "type": "value",
  },
  "yAxis": Object {
    "axisLabel": Object {
      "formatter": [Function],
      "interval": 0,
    },
    "axisPointer": Object {
      "label": Object {
        "formatter": [Function],
      },
      "show": true,
      "type": "none",
    },
    "axisTick": Object {
      "show": true,
    },
    "boundaryGap": true,
    "name": "User",
    "nameGap": 50,
    "nameLocation": "center",
    "splitLine": Object {
      "show": false,
    },
    "type": "category",
  },
}
`;
